lib/ansible/module_utils/oneview.py

```diff
@@ -0,0 +1,438 @@
+
+        for i, val in enumerate(resource1):
+            if isinstance(val, collections.Mapping):
+                # change comparison function to compare dictionaries
+                if not self.compare(val, resource2[i]):
+                    self.module.log("resources are different. " + debug_resources)
+                    return False
+            elif isinstance(val, list):
+                # recursive call
+                if not self.compare_list(val, resource2[i]):
+                    self.module.log("lists are different. " + debug_resources)
+                    return False
+            elif self._standardize_value(val) != self._standardize_value(resource2[i]):
+                self.module.log("values are different. " + debug_resources)
+                return False
+
+        # no differences found
+        return True
+
+    def _str_sorted(self, obj):
+        if isinstance(obj, collections.Mapping):
+            return json.dumps(obj, sort_keys=True)
+        else:
+            return str(obj)
+
+    def _standardize_value(self, value):
+        """
+        Convert value to string to enhance the comparison.
+
+        :arg value: Any object type.
+
+        :return: str: Converted value.
+    """
+        if isinstance(value, float) and value.is_integer():
+            # Workaround to avoid erroneous comparison between int and float
+            # Removes zero from integer floats
+            value = int(value)
+
+        return str(value)
+
+    def merge_list_by_key(self, original_list, updated_list, key, ignore_when_null=[]):
+        """
+        Merge two lists by the key. It basically:
+
+        1. Adds the items that are present on updated_list and are absent on original_list.
+
+        2. Removes items that are absent on updated_list and are present on original_list.
+
+        3. For all items that are in both lists, overwrites the values from the original item by the updated item.
+
+        :arg list original_list: original list.
+        :arg list updated_list: list with changes.
+        :arg str key: unique identifier.
+        :arg list ignore_when_null: list with the keys from the updated items that should be ignored in the merge,
+            if its values are null.
+        :return: list: Lists merged.
+"""
+        if not original_list:
+            return updated_list
+
+        items_map = collections.OrderedDict([(i[key], i.copy()) for i in original_list])
+
+        merged_items = collections.OrderedDict()
+
+        for item in updated_list:
+            item_key = item[key]
+            if item_key in items_map:
+                for ignored_key in ignore_when_null:
+                    if ignored_key in item and not item[ignored_key]:
+                        item.pop(ignored_key)
+                merged_items[item_key] = items_map[item_key].copy()
+                merged_items[item_key].update(item)
+            else:
+                merged_items[item_key] = item.copy()
```

Comment:
I see that we're still doing copies of the item here twice, once when we enter it into items map at the beginning of hte function and once when we enter it into merged_items here.  Is that needed?